---
template: meglab.html
title: Home
---

# 

## Who we are
The Microbial Ecology Group (MEG) is an interdisciplinary group of scientists from multiple institutions that conduct collaborative research addressing the issues of microbial ecology in animal, public, and environmental health. Lead scientists for MEG hail from (alphabetically): Colorado State University, Texas A&amp;M University, University of Florida, University of Minnesota, and West Texas A&amp;M University.


## What we do
Microbial ecologyÂ is the study of diversity, distribution, and abundance of microorganisms, their interactions with each other and the biotic and abiotic features of their environments, and the effects that they have on ecosystems[^1]. It is increasingly recognized that  health states are affected by complex interactions within microbial communities (the microbiome) and between the microbiome and hosts can have dramatic impacts on health and disease.

MEG team members are experts in agricultural and food production systems, veterinary medicine, public health, molecular biology, microbiology and food safety, host genomics, ecosystem health, epidemiology, computational biology, bioinformatics, and advanced statistics. The MEG team integrate their unique talents and expertise to collaborative address the critically important research regarding antimicrobial resistance, microbial ecology, host-microbe interactions, and infectious disease pathogenesis as these affect animals, public health, and ecosystem health.

## Open Science - Sharing Your Research with the World:

MEG embraces the philosophy of open science to promote transparency, and improve repeatability and efficiency in research. In line with this principle, we are sharing the tools that we have developed through our research to facilitate metagenomic investigations of antimicrobial resistance using genomic sequencing and high-throughput computational analysis.

### [MEGARes](./megares/index.md)
The MEGARes v3.0 database contains sequence data for nearly 9,000 hand-curated antimicrobial resistance genes accompanied by an annotation structure that is optimized for use with high throughput sequencing. The acyclical annotation graph of MEGARes allows for accurate, count-based, hierarchical statistical analysis of resistance at the population level, much like microbiome analysis, and is also designed to be used as a training database for the creation of statistical classifiers.

### [AMR++ Pipeline](./amrplusplus/index.md)
AMR++ is a bioinformatic pipeline meant to aid in the analysis of raw sequencing reads to  characterize the profile of antimicrobial resistance genes, or resistome. AMR++ was developed to work in conjuction with the the MEGARes database and its accompanying acyclical annotation structure that is optimized for use with high throughput sequencing and metagenomic analysis.  AMR++ v3.0 adds a new feature for high-throughput verification of resistance-conferring SNPs in relevant gene accessions (ARGs).

[^1]: Gray, N. D. &amp; Head, I. M. in Encyclopedia of Ecology (eds S.E. Jorgensen &amp; B.D. Fath) 2357-2368 (Elsevier Science, 2008).




